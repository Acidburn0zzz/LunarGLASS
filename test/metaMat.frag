#version 300 es
precision highp float;

struct S1 {
    uint i;
    mat2x3 S1m;
};

struct S2 {
    uint i;
    mat2x3 S2m;
};

struct Si1 {
    float f;
    S1 s1;
};

struct Si2 {
    float g;
    S2 s2;
};

uniform BNA {
    mat3x4 BNAm;
    Si1 si1;
} BNAI;

layout (row_major) uniform BNB {
    mat3 BNBm3;
    Si1 si1;
    layout(column_major) Si2 si21;
    layout(row_major) Si2 si22;
    layout(column_major) mat4 BNBm41;
    mat4 BNBm42;
} BNBI;

void main()
{
}