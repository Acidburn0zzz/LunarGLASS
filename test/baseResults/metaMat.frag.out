
Top IR:
; ModuleID = 'Glslang'

%BNA = type { [3 x <4 x float>], %Si1 }
%Si1 = type { float, %S1 }
%S1 = type { i32, [2 x <3 x float>] }
%BNB = type { [3 x <3 x float>], %Si1, %Si2, %Si2, [4 x <4 x float>], [4 x <4 x float>] }
%Si2 = type { float, %S2 }
%S2 = type { i32, [2 x <3 x float>] }

@BNAI = external addrspace(2) constant %BNA
@BNBI = external addrspace(2) constant %BNB

define fastcc void @main() {
entry:
  br label %mainBody

mainBody:                                         ; preds = %entry
  br label %stage-epilogue

stage-epilogue:                                   ; preds = %mainBody
  br label %stage-exit

stage-exit:                                       ; preds = %stage-epilogue
  ret void
}

!gla.entrypoint = !{!0}
!gla.uniforms = !{!1, !13}
!gla.noStaticUse = !{!1, !13}

!0 = metadata !{metadata !"main", i32 15}
!1 = metadata !{metadata !"BNAI", i32 13, %BNA* @BNAI_typeProxy, metadata !2, metadata !3}
!2 = metadata !{i32 6, i32 0, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!3 = metadata !{metadata !"BNA", metadata !2, metadata !"BNAm", metadata !4, metadata !"si1", metadata !6}
!4 = metadata !{metadata !"", metadata !5}
!5 = metadata !{i32 3, i32 3, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!6 = metadata !{metadata !"Si1", metadata !7, metadata !"f", metadata !8, metadata !"s1", metadata !10}
!7 = metadata !{i32 4, i32 0, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!8 = metadata !{metadata !"", metadata !9}
!9 = metadata !{i32 0, i32 3, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!10 = metadata !{metadata !"S1", metadata !7, metadata !"i", metadata !11, metadata !"S1m", metadata !4}
!11 = metadata !{metadata !"", metadata !12}
!12 = metadata !{i32 1, i32 2, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!13 = metadata !{metadata !"BNBI", i32 13, %BNB* @BNBI_typeProxy, metadata !2, metadata !14}
!14 = metadata !{metadata !"BNB", metadata !2, metadata !"BNBm3", metadata !15, metadata !"si1", metadata !17, metadata !"si21", metadata !19, metadata !"si22", metadata !21, metadata !"BNBm41", metadata !4, metadata !"BNBm42", metadata !15}
!15 = metadata !{metadata !"", metadata !16}
!16 = metadata !{i32 2, i32 3, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!17 = metadata !{metadata !"Si1", metadata !7, metadata !"f", metadata !8, metadata !"s1", metadata !18}
!18 = metadata !{metadata !"S1", metadata !7, metadata !"i", metadata !11, metadata !"S1m", metadata !15}
!19 = metadata !{metadata !"Si2", metadata !7, metadata !"g", metadata !8, metadata !"s2", metadata !20}
!20 = metadata !{metadata !"S2", metadata !7, metadata !"i", metadata !11, metadata !"S2m", metadata !4}
!21 = metadata !{metadata !"Si2", metadata !7, metadata !"g", metadata !8, metadata !"s2", metadata !22}
!22 = metadata !{metadata !"S2", metadata !7, metadata !"i", metadata !11, metadata !"S2m", metadata !15}


Bottom IR:
; ModuleID = 'Glslang'

define fastcc void @main() {
entry:
  br label %stage-epilogue

stage-epilogue:                                   ; preds = %entry
  br label %stage-exit

stage-exit:                                       ; preds = %stage-epilogue
  ret void
}

!gla.entrypoint = !{!0}
!gla.uniforms = !{!1, !13}
!gla.noStaticUse = !{!1, !13}

!0 = metadata !{metadata !"main", i32 15}
!1 = metadata !{metadata !"BNAI", i32 13, %BNA* @BNAI_typeProxy, metadata !2, metadata !3}
!2 = metadata !{i32 6, i32 0, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!3 = metadata !{metadata !"BNA", metadata !2, metadata !"BNAm", metadata !4, metadata !"si1", metadata !6}
!4 = metadata !{metadata !"", metadata !5}
!5 = metadata !{i32 3, i32 3, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!6 = metadata !{metadata !"Si1", metadata !7, metadata !"f", metadata !8, metadata !"s1", metadata !10}
!7 = metadata !{i32 4, i32 0, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!8 = metadata !{metadata !"", metadata !9}
!9 = metadata !{i32 0, i32 3, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!10 = metadata !{metadata !"S1", metadata !7, metadata !"i", metadata !11, metadata !"S1m", metadata !4}
!11 = metadata !{metadata !"", metadata !12}
!12 = metadata !{i32 1, i32 2, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!13 = metadata !{metadata !"BNBI", i32 13, %BNB* @BNBI_typeProxy, metadata !2, metadata !14}
!14 = metadata !{metadata !"BNB", metadata !2, metadata !"BNBm3", metadata !15, metadata !"si1", metadata !17, metadata !"si21", metadata !19, metadata !"si22", metadata !21, metadata !"BNBm41", metadata !4, metadata !"BNBm42", metadata !15}
!15 = metadata !{metadata !"", metadata !16}
!16 = metadata !{i32 2, i32 3, i32 1024, null, i32 -1, i32 0, i32 -1, i32 0, i32 -1}
!17 = metadata !{metadata !"Si1", metadata !7, metadata !"f", metadata !8, metadata !"s1", metadata !18}
!18 = metadata !{metadata !"S1", metadata !7, metadata !"i", metadata !11, metadata !"S1m", metadata !15}
!19 = metadata !{metadata !"Si2", metadata !7, metadata !"g", metadata !8, metadata !"s2", metadata !20}
!20 = metadata !{metadata !"S2", metadata !7, metadata !"i", metadata !11, metadata !"S2m", metadata !4}
!21 = metadata !{metadata !"Si2", metadata !7, metadata !"g", metadata !8, metadata !"s2", metadata !22}
!22 = metadata !{metadata !"S2", metadata !7, metadata !"i", metadata !11, metadata !"S2m", metadata !15}
#version 300 es
// LunarGOO output
precision mediump float; // this will be almost entirely overridden by individual declarations
struct S1 {
	mediump uint i;
	highp mat2x3 S1m;
};
struct Si1 {
	highp float f;
	S1 s1;
};
struct S2 {
	mediump uint i;
	highp mat2x3 S2m;
};
struct Si2 {
	highp float g;
	S2 s2;
};
uniform BNA {
	highp mat3x4 BNAm;
	Si1 si1;
} BNAI;
uniform BNB {
	layout(row_major) highp mat3 BNBm3;
	Si1 si1;
	Si2 si21;
	Si2 si22;
	highp mat4 BNBm41;
	layout(row_major) highp mat4 BNBm42;
} BNBI;

void main()
{
	
}

#version 300 es
// LunarGOO output
precision mediump float; // this will be almost entirely overridden by individual declarations
struct S1 {
	mediump uint i;
	highp mat2x3 S1m;
};
struct Si1 {
	highp float f;
	S1 s1;
};
struct S2 {
	mediump uint i;
	highp mat2x3 S2m;
};
struct Si2 {
	highp float g;
	S2 s2;
};
uniform BNA {
	highp mat3x4 BNAm;
	Si1 si1;
} BNAI;
uniform BNB {
	layout(row_major) highp mat3 BNBm3;
	Si1 si1;
	Si2 si21;
	Si2 si22;
	highp mat4 BNBm41;
	layout(row_major) highp mat4 BNBm42;
} BNBI;

void main()
{
	
}

